lib/foreman_puppet_enc/engine.rb

```diff
@@ -13,17 +22,55 @@ class Engine < ::Rails::Engine
     initializer 'foreman_puppet_enc.register_plugin', before: :finisher_hook do |_app|
       Foreman::Plugin.register :foreman_puppet_enc do
         requires_foreman '>= 2.0.0'
-
         # Add Global JS file for extending foreman-core components and routes
         register_global_js_file 'fills'
 
         # Add permissions
-        # security_block :foreman_puppet_enc do
-        # end
+        security_block :foreman_puppet_enc do
+          permission :view_config_groups, { :config_groups => %i[index auto_complete_search welcome],
+                                            :"api/v2/config_groups" => %i[index show],
+                                            :'foreman_puppet_enc/react' => [:index] }
+          permission :create_config_groups, { :config_groups => %i[new create],
+                                              :"api/v2/config_groups" => [:create] }
+          permission :edit_config_groups, { :config_groups => %i[edit update],
+                                            :"api/v2/config_groups" => [:update] }
+          permission :destroy_config_groups, { :config_groups => [:destroy],
+                                               :"api/v2/config_groups" => [:destroy] }
+        end
 
-        # add menu entry
+        role 'ForemanPuppetEnc', [:view_foreman_puppet_enc]
```

Comment:
Do we want extra role for PuppetEnc handling?

Reply:
I think we should, what do you think?

Reply:
I would keep the roles as they are now in core and there is no special role for managing puppet :thinking:
If to have such role, I would say it should be a role `ConfigurationManager` or something like that, but otherwise I'd keep things simple.